### PR TITLE
[CommunityToolkit.Maui.Options] Add `DefaultConvertReturnValue` and `DefaultConvertBackReturnValue`

### PR DESCRIPTION
**Pull request by brminnick**
> Note: This PR targets the `maui-options` branch, not the `main` branch. In other words, when this PR is merged, it will not be merged to `main`; it will be merged into PR #616.

Hey @VladislavAntonyuk!

I took a stab at adding `DefaultConvertReturnValue` and `DefaultConvertBackReturnValue` (as [discussed here](https://github.com/CommunityToolkit/Maui/pull/616#issuecomment-1244093116)) and wanted to share it with you before we add it to your PR, #616.

## Explanation

I just realized that we should have two different properties in `BaseConverter` and `ICommunityToolkitValueConverter` to allow the user to set different values for `TFrom` and `TTo`.

### ICommunityToolkitValueConverter

```cs
public interface ICommunityToolkitValueConverter : IValueConverter
{
  object? DefaultConvertReturnValue { get; }
  object? DefaultConvertBackReturnValue { get; }
  Type FromType { get; }
  Type ToType { get; }

   // ...
}
```

### BaseConverter

```cs
public class BaseConverter<TFrom, TTo, TParam> : ValueConverterExtension, ICommunityToolkitValueConverter
{
  public abstract TTo DefaultConvertReturnValue { get; set; }
  public abstract TFrom DefaultConvertBackReturnValue { get; set; }
  public Type FromType { get; } = typeof(TFrom);
  public Type ToType { get; } = typeof(TTo);

  object? ICommunityToolkitValueConverter.DefaultConvertReturnValue => DefaultConvertReturnValue;
  object? ICommunityToolkitValueConverter.DefaultConvertBackReturnValue => DefaultConvertBackReturnValue;

  // ...

  object? ICommunityToolkitValueConverter.ConvertBack(object? value, Type targetType, object? parameter, CultureInfo? culture)
  {
    try
    {
      ValidateTargetType<TFrom>(targetType);
      
      var converterParameter = ConvertParameter<TParam>(parameter);
      var converterValue = ConvertValue<TTo>(value);
      
      return ConvertBackTo(converterValue, converterParameter, culture);
    }
    catch (Exception ex) when (Options.ShouldSuppressExceptionsInConverters)
    {
      Debug.WriteLine(ex);
      return DefaultConvertBackReturnValue;
    }
  }

  object? ICommunityToolkitValueConverter.Convert(object? value, Type targetType, object? parameter, CultureInfo? culture)
  {
    try
    {
      ValidateTargetType<TTo>(targetType);
      
      var converterParameter = ConvertParameter<TParam>(parameter);
      var converterValue = ConvertValue<TFrom>(value);
      
      return ConvertFrom(converterValue, converterParameter, culture);
    }
    catch (Exception ex) when (Options.ShouldSuppressExceptionsInConverters)
    {
      Debug.WriteLine(ex);
      return DefaultConvertReturnValue;
    }
  }
}
```

### CODE REVIEW

File: src/CommunityToolkit.Maui/Converters/EnumToIntConverter.shared.cs [hunk 1]
```diff
@@ -9,7 +9,10 @@ namespace CommunityToolkit.Maui.Converters;
 public class EnumToIntConverter : BaseConverter<Enum, int, Type>
 {
 	/// <inheritdoc/>
-	public override int DefaultReturnValue { get; set; } = 0;
+	public override int DefaultConvertReturnValue { get; set; } = 0;
+
+	/// <inheritdoc/>
+	public override Enum DefaultConvertBackReturnValue { get; set; } = (Enum)Enum.ToObject(typeof(Enum), 0);
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Converters/EnumToIntConverter.shared.cs:15]: Matches the return statement of `ConvertBack`:

```cs
return (Enum)Enum.ToObject(parameter, value);
```

**Comment by VladislavAntonyuk** [comment]
Absolutely agree!

File: src/CommunityToolkit.Maui/Converters/BaseConverterOneWay.shared.cs [hunk 2]
```diff
@@ -9,6 +9,13 @@ namespace CommunityToolkit.Maui.Converters;
 /// <typeparam name="TTo">Type of the output value</typeparam>
 public abstract class BaseConverterOneWay<TFrom, TTo> : BaseConverter<TFrom, TTo>
 {
+	/// <inheritdoc/>
+	public override TFrom DefaultConvertBackReturnValue
+	{
+		get => throw new NotSupportedException($"{nameof(DefaultConvertBackReturnValue)} not used for ${nameof(BaseConverterOneWay<TFrom, TTo>)}");
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Converters/BaseConverterOneWay.shared.cs:15]: you know English better than I do. should it be **is** not used?

**Review by VladislavAntonyuk — APPROVED** [review]

File: src/CommunityToolkit.Maui/Converters/BaseConverterOneWay.shared.cs (hunk 2 above)
Comment (brminnick) [reply src/CommunityToolkit.Maui/Converters/BaseConverterOneWay.shared.cs:15]: Yup! Fixed ✅

Your English is clearly better than mine 😆🤦‍♂️

**Pull request merged**